library/core/src/str/mod.rs

```diff
@@ -1297,7 +1297,7 @@ impl str {
     #[inline]
     pub fn rfind<'a, P>(&'a self, pat: P) -> Option<usize>
     where
-        P: Pattern<'a, Searcher: ReverseSearcher<'a>>,
+        P: Pattern<Searcher<'a>: ReverseSearcher<'a>>,
```

Comment:
Would something like this help?
```suggestion
        P: Pattern,
        for<'a> P::Searcher<'a>: ReverseSearcher<'a>,
```

Reply:
It should work: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=af8eb63e3403dbfaaea02f2246a68bb6